translation-service/src/main/java/eu/europeana/api/translation/service/PangeanicTranslationService.java

```diff
@@ -0,0 +1,231 @@
+    cm.setDefaultMaxPerRoute(PangeanicTranslationUtils.MAX_CONNECTIONS_PER_ROUTE);
+    cm.setDefaultSocketConfig(
+        SocketConfig.custom().setSoKeepAlive(true).setSoTimeout(3600000).build());
+    // SocketConfig socketConfig =
+    // SocketConfig.custom().setSoKeepAlive(true).setSoTimeout(3600000).build(); //We need to set
+    // socket keep alive
+    translateClient = HttpClients.custom().setConnectionManager(cm).build();
+    LOG.info("Pangeanic translation service is initialized with translate Endpoint - {}",
+        getExternalServiceEndPoint());
+  }
+
+  /**
+   * target language should be English for Pangeanic Translations and validate the source language
+   * with list of supported languages
+   *
+   * @param srcLang source language of the data to be translated
+   * @param targetLanguage target language in which data has to be translated
+   * @return
+   */
+  @Override
+  public boolean isSupported(String srcLang, String targetLanguage) {
+    if(srcLang == null) {
+      //automatic language detection
+      return isTargetSupported(targetLanguage);  
+    }
+    return PangeanicLanguages.isLanguagePairSupported(srcLang, targetLanguage);
+  }
+
+  private boolean isTargetSupported(String targetLanguage) {
+    return PangeanicLanguages.isTargetLanguageSupported(targetLanguage);
+  }
+
+
+  @Override
+  public List<String> translate(List<String> texts, String targetLanguage, String sourceLanguage) throws TranslationException {
+    try {
+      if (sourceLanguage == null) {
+        // In this case source language is the hint. The texts passed will be sent for
+        // lang-detection first and later will translated
+        return translateWithLangDetect(texts, targetLanguage, sourceLanguage);
+      }
+      HttpPost post = PangeanicTranslationUtils.createTranslateRequest(getExternalServiceEndPoint(),
+          texts, targetLanguage, sourceLanguage, "");
+      return PangeanicTranslationUtils.getResults(texts,
+          sendTranslateRequestAndParse(post, sourceLanguage), false);
+    } catch (JSONException | IOException e) {
+      throw new TranslationException(e.getMessage());
+    }
+  }
+
+  @Override
+  public List<String> translate(List<String> texts, String targetLanguage)
+      throws TranslationException {
+    return translate(texts, targetLanguage, null);
+  }
+  
+  
+  /**
+   * Translates the texts with no source language. First a lang detect request is sent to identify
+   * the source language Later translations are performed
+   *
+   * @param texts
+   * @param targetLanguage
+   * @return
+   * @throws TranslationException
+   */
+  private List<String> translateWithLangDetect(List<String> texts, String targetLanguage,
+      String langHint) throws TranslationException {
+    try {
+      if (langDetectService == null) {
+        throw new TranslationException("No langDetectService configured!");
+      }
+      List<String> detectedLanguages;
+      try {
```

Comment:
extract this nested try-catch to a seperate method

Reply:
will do